test/uController.SourceGenerator.Tests/IntegrationTests.cs

```diff
@@ -807,7 +808,85 @@ public static object[][] FromServiceActions
         }
     }
 
-    private async Task<RequestDelegate> GetRequestDelegate(string source, IServiceProvider? serviceProvider = null)
+    public static object[][] AcceptsMetadataActions
+    {
+        get
+        {
+            var implicitFromFormFile = 
+            $$"""
+            app.MapPost("/fileupload", (IFormFile file) =>
+            {
+                return $"Uploaded {file.Name}";
+            });
+            """;
+
+            var implicitFromFormCollection =
+            $$"""
+            app.MapPost("/formpost", (IFormCollection formCollection) =>
+            {
+                return $"Uploaded {formCollection.Count} files";
+            });
+            """;
+
+            var explicitFromFormFile = 
+            $$"""
+            app.MapPost("/fileupload", ([FromForm] IFormFile file) =>
+            {
+                return $"Uploaded {file.Name}";
+            });
+            """;
+
+            var explicitFromFormCollection =
+            $$"""
+            app.MapPost("/formpost", ([FromForm] IFormCollection formCollection) =>
+            {
+                return $"Uploaded {formCollection.Count} files";
+            });
+            """;
+
+            return new[]
+            {
+                new[] { ImplicitFromBodyActions[0][0], typeof(Todo), new[] { "application/json" }},
+                new[] { ImplicitFromBodyActions[1][0], typeof(ITodo), new[] { "application/json" }},
+                new[] { ImplicitFromBodyActions[2][0], typeof(TodoStruct), new[] { "application/json" }},
+                new[] { ExplicitFromBodyActions[0][0], typeof(Todo), new[] { "application/json" }},
```

Comment:
This look fragile. Is there a better way to do this without indexing into this? Maybe this is the source of truth and the other dataset uses this one and ignores the return type and content type?